content/docs/tasks/traffic-management/app-health-check/index.md

```diff
@@ -114,27 +128,111 @@ Remove the mutual TLS policy and corresponding destination rule added in the ste
 
 ## Liveness and readiness probes with HTTP request option
 
-This section shows how to configure health checking with the HTTP request option.
+This section shows how to configure health checking with the HTTP request
+option. Firstly it discusses the case when the service is using same port
+for regular traffic and health checking. Then it discusses the case with
+different port.
 
-### Mutual TLS is disabled
+### Mutual TLS is disabled (same port)
 
-Run this command to deploy [liveness-http]({{< github_file >}}/samples/health-check/liveness-http.yaml) in the default namespace:
+Run this command to deploy [liveness-http]({{< github_file >}}/samples/health-check/liveness-http-same-port.yaml)
+in the default namespace:
+
+{{< text bash >}}
+$ kubectl apply -f <(istioctl kube-inject -f @samples/health-check/liveness-http-same-port.yaml@)
+{{< /text >}}
+
+Wait for a minute and check the pod status to make sure the liveness probes
+work with '0' in the 'RESTARTS' column.
+
+{{< text bash >}}
+$ kubectl get pod
+NAME                             READY     STATUS    RESTARTS   AGE
+liveness-http-975595bb6-5b2z7c   2/2       Running   0           1m
+{{< /text >}}
+
+### Mutual TLS is enabled (same port)
+
+In this case, health checking will break if the service is enforced with mutual
+TLS and the reason is that the kubelet does not have Istio-issued client certificate
+to talk to the service. As a result, you can only use
+[`PERMISSIVE` mode](/docs/tasks/security/mtls-migration) for this service, which
+accepts both http and mutual TLS traffic at the same time. Please keep in mind
+in this case, mutual TLS is not enforced and the recommended way for health
+checking is to use a different port, which is discussed in the next section.
```

Comment:
The whole paragraph is unclear, please consider:
In this case, health checking breaks if mTLS is enabled for the service
because the kubelet does not have an Istio-issued client certificate
for the service. 

As a result, you can only use the `PERMISSIVE` mode for this service. This mode
accepts both HTTP and mutual TLS traffic at the same time you can find more information in our [mTLS documentation](/docs/tasks/security/mtls-migration). Keep in mind, mutual TLS is not enforced and we recommend you use a different port for health checking as discussed in the next section.